packages/devtools_app/integration_test/run_tests.dart

```diff
@@ -56,25 +53,29 @@ void main(List<String> args) async {
 
     for (final testFile in testFiles) {
       final testTarget = testFile.path;
-      _maybeAddOfflineArgument(modifiableArgs, testTarget);
-      _maybeAddExperimentsArgument(modifiableArgs, testTarget);
-      final testRunnerArgs = TestArgs([
-        ...modifiableArgs,
-        '${TestArgs.testTargetArg}$testTarget',
-      ]);
-      await runFlutterIntegrationTest(testRunnerArgs);
+      modifiableArgs.add('${TestArgs.testTargetArg}$testTarget');
+      await _runTest(modifiableArgs, testTarget);
     }
   }
 }
 
-void _maybeAddOfflineArgument(List<String> args, String testTarget) {
-  if (testTarget.startsWith(_offlineIndicator)) {
-    args.add(TestArgs.offlineArg);
+Future<void> _runTest(List<String> modifiableArgs, String testFilePath) async {
+  _maybeAddOfflineArgument(modifiableArgs, testFilePath);
+
+  final inFileArgs = InFileArgs(testFilePath);
+  final cliArgs = TestArgs(modifiableArgs);
```

Comment:
nit: maybe these classes  (and variables) are more aptly named TestFileArgs and TestRunnerArgs